pipeline/r53update.py

```diff
@@ -0,0 +1,278 @@
+    def fmt(s):
+        r = s.format(**format_args)
+        logger.debug(f"formatted {s!r} -> {r!r}")
+        return r
+    zone_name = fmt(args.zone_format)
+    if not zone_name.endswith('.'):
+        zone_name += '.'
+    node_ips = args.nodes
+    if args.debug:
+        logger.setLevel(logging.DEBUG)
+    else:
+        logger.setLevel(logging.INFO)
+
+    if node_ips:
+        if args.confirm_remove:
+            parser.error("--confirm-remove cannot be given with nodes")
+    else:
+        if not args.confirm_remove:
+            parser.error("no node IPs given, will only remove existing ones; "
+                         "add --confirm-remove to confirm this")
+
+    logger.debug("creating Route 53 client")
+    session_kwargs = dict()
+    if args.aws_profile:
+        session_kwargs.update(profile_name=args.aws_profile)
+    session = boto3.Session(**session_kwargs)
+    r53 = session.client('route53')
+
+    logger.debug(f"retrieving zone ID for zone {zone_name!r}")
+    zones = {zone['Name']: zone for zone in list_hosted_zones(r53)}
+    parent_zone, subzone_name = _find_parent(zones, zone_name)
+    logger.debug(f"parent {parent_zone['Name']!r}, subzone {subzone_name!r}")
+    if subzone_name is None:
+        # not found, not even potential parent
+        logger.error(f"zone {zone_name!r} is not found")
+        return 1
+    elif subzone_name == '':
+        # exact match
+        zone = parent_zone
+    elif not args.create_zone:
+        logger.error(f"zone {zone_name!r} is not found; "
+                     f"add --create-zone to create in {parent_zone['Name']!r}")
+        return 1
+    else:
+        logger.info(f"zone {zone_name!r} is not found, "
+                    f"creating in {parent_zone['Name']!r}")
+        zone = _create_subzone(r53, zone_name, parent_zone['Id'])
+    zone_id = zone['Id']
+
+    batch = []
+    rrtype_by_version = {4: 'A', 6: 'AAAA'}
+
+    shard_label = fmt(args.shard_format)
+    leader_label = fmt(args.leader_format)
+
+    def schedule_removal(label):
+        for rrtype in rrtype_by_version.values():
+            name = f'{label}.{zone_name}'
+            for old_rrset in get_rrsets(r53, zone_id, name, rrtype):
+                logger.info(f"scheduling removal of {name!r} {rrtype} RRs")
+                batch.append({
+                    'Action': 'DELETE',
+                    'ResourceRecordSet': old_rrset,
+                })
+
+    schedule_removal(shard_label)
+    schedule_removal(leader_label)
+
+    # classify by rrtype
+    nodes = {}
+    for node_ip in node_ips:
+        nodes.setdefault(node_ip.version, []).append(node_ip)
+
+    for version, addrs in nodes.items():
+        logger.debug(f"adding {len(addrs)} IPv{version} nodes")
+        assert addrs
+        batch.append({
+            'Action': 'CREATE',
+            'ResourceRecordSet': {
+                'Name': f'{shard_label}.{zone_name}',
+                'Type': rrtype_by_version[version],
+                'TTL': 60,
```

Comment:
is it too short? 60?

Reply:
Yeah this is so that when you restart the network and update the DNS, existing cached entries across the network will expire within a minute and clients will start using the new network.